File: Algorithm/탐색/bfs.py
#queue 자료구조

#큐(queue) 구현을 위해 deque 라이브러리 사용
#queue = deque()

#queue.append(5)
#queue.append(2)
#queue.append(3)
#queue.append(7)
#queue.popleft()
#queue.append(1)
#queue.append(4)
#queue.popleft()

#print(queue)
#queue.reverse()
#print(queue)

from collections import deque

def bfs(graph,start,visited):
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v,end = ' ')
        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True

graph = [
    [],
    [2, 3, 8],
    [1, 7],
    [1, 4, 5],
    [3,5],
    [3,4],
    [7],
    [2,6,8],
    [1,7]
]

visited = [False] * 9

bfs(graph,1,visited)

File: Kakao test/Commerce/src/graph/Bfs.java
package graph;

import java.util.LinkedList;
import java.util.Queue;

class Node{
    int x;
    int y;

    Node(int x, int y) {
        this.x = x;
        this.y = y;
    }
}

public class Bfs {
    public static void main(String[] args) {

    }

    public static void bfs(int start, int end) {
        Queue<Node> queue = new LinkedList<>();
        queue.add(new Node(start, end));
        int[] dx = {0,1,0,-1};
        int[] dy = {1,0,-1,0};

        while(!queue.isEmpty()) {
            Node node = queue.poll();

            for(int i = 0 ; i < 4 ; i ++) {
                int xx = node.x + dx[i];
                int yy = node.y + dy[i];
                if(0 <= xx && xx < 10 && 0 <= yy && yy < 10) {
                    queue.add(new Node(xx,yy));
                }
            }
        }
    }
}
